### PR TITLE
Adds part 4 of the pipeline

### PR DESCRIPTION
**Pull request by Vlad-Dembrovskyi**
## Adds processes:
- [x] create_sparce_GRM
- [x] fit_null_GLMM
- [x] SPA_tests
- [x] concat_output

## Fixes:
- #9 (and duplicate #10) : Fix tuple sizes discordance in some edge cases
- fetching proper plink files by basename by changing `getSimpleName()` to `getName()` so it tolerates dots in plink file basename.
- Changes error strategy from `ignore` to `terminate` so we can catch the error on the very first failed task.
- All process-wise configs like `withLabel` are moved from `params` to `procss` scope in **aws.config**  file so they now actually work properly.

### To test
The current version of pipeline (that includes part 4) is not testable at the moment outside of VDI, so cannot be tested locally or on main CloudOS. Better test data is required for that. The GEL cloudOS test run on subset of real data will be added in ~15 hours when it is done.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -275,7 +289,7 @@ Channel
 ch_mac_categories = Channel.from(params.mac_categories)
 
 ch_sparse_plink_files = Channel.from(params.sparse_plink_name)
-                                .map { name -> [file(name).getSimpleName(),
+                                .map { name -> [file(name).getName(),
```
Comment (cgpu) [inline main.nf:292]: Not sure how this works differently, but this is typically written as follows as well just fyi:

```Suggestion
                                .map { name -> [file(name.baseName),
```

See example:
https://github.com/cgpu/ion-somatic-variant-calling/blob/85744c97d57620a0e42a96a92cf1479537979637/main.nf#L128

File: main.nf [hunk 2]
```diff
@@ -275,7 +289,7 @@ Channel
 ch_mac_categories = Channel.from(params.mac_categories)
 
 ch_sparse_plink_files = Channel.from(params.sparse_plink_name)
-                                .map { name -> [file(name).getSimpleName(),
```
Comment (cgpu) [inline main.nf:278]: Equivalent:

```Suggestion
                                .map { name -> [file(name.simpleName,	
```

File: main.nf [hunk 3]
```diff
@@ -999,8 +1017,231 @@ process create_plink_set {
...
+    relatedness_cutoff_for_sparseGRM = params.relatedness_cutoff_for_sparseGRM
+    num_random_marker_for_sparse_kin = params.num_random_marker_for_sparse_kin
+    """
+    createSparseGRM.R \
+         --plinkFile=${plink_files_for_GRM} \
+         --nThreads=${task.cpus} \
+         --outputPrefix=sparse_GRM \
+         --relatednessCutoff=${relatedness_cutoff_for_sparseGRM} \
+         --numRandomMarkerforSparseKin=${num_random_marker_for_sparse_kin} \
+         --memoryChunk 2
```
Comment (cgpu) [inline main.nf:1049]: Should the value stay 2 or be exposed to the user?

File: main.nf [hunk 4]
```diff
@@ -999,8 +1017,231 @@ process create_plink_set {
...
+    IsOutputNinCaseCtrl = params.IsOutputNinCaseCtrl
+    condition = params.condition
+    maxMAFforGroupTest = params.maxMAFforGroupTest
+    cateVarRatioMinMACVecExclude = params.cateVarRatioMinMACVecExclude
+    cateVarRatioMaxMACVecInclude = params.cateVarRatioMaxMACVecInclude
+    IsSingleVarinGroupTest = params.IsSingleVarinGroupTest
+    dosageZerodCutoff = params.dosageZerodCutoff
+    IsOutputPvalueNAinGroupTestforBinary = params.IsOutputPvalueNAinGroupTestforBinary
+    IsAccountforCasecontrolImbalanceinGroupTest = params.IsAccountforCasecontrolImbalanceinGroupTest
+    method = params.method
```
Comment (cgpu) [inline main.nf:1165]: I know we are retaining WDL variable names but CAPS break the syntax highlighting and now we have half/half snakecase camelCase.

I would not mind making everything into `snake_case` but will not insist if there is a strong reason to keep as camelCase.

**Review by cgpu — COMMENTED** [review]
Looks good generally.

Thoughts:

- Not very fond of mix of cam and snakecase
- Some channels are a bit flaky, eg the recreation on the fly of file names with +suffix
- We will need to revisit the parameters to add description once we review the GEL docs.

File: main.nf (hunk 4 above)
Comment (Vlad-Dembrovskyi) [reply main.nf:1120]: These parameters are named the same as SAIGE parameters. People that will be using AVT pipeline will immediately know which AVT parameters are SAIGE parameters just by looking at them. SAIGE is the core software of the pipeline.

**Review by cgpu — APPROVED** [review]
Looks good, thanks for adding extra parameters documentation 🏎️

File: main.nf (hunk 3 above)
Comment (Vlad-Dembrovskyi) [reply main.nf:1004]: stay 2

**Pull request merged**